Automattic/woocommerce-payments
# Merge split UPE tests with main gateway tests

Closes https://github.com/Automattic/woocommerce-payments/issues/7953

#### Changes proposed in this Pull Request
This PR merges three unit test suites into one, making `test-class-wc-payment-gateway-wcpay.php` the only test suite that covers everything. While combining, several improvements were made to the tests. The most significant one is we don't mock the unit under tests now.

Additionally, this PR removes the duplicates from E2E tests. Previously, we had multiple test suites checking e.g. saving cards and each test suite was doing it under different setting e.g. UPE or legacy card. Since we streamlined them now, there's no need to have duplicated tests either.

Tests that, in my view, need further analysis to comprehend their outcomes are accompanied by my comments. These comments detail whether the test was removed, relocated to a file, or modified. If a test doesn't have a comment, this means it was simply moved to the main gateway test suite w/o any big modifications, but of course feel free to raise any questions you might have. 









#### Testing instructions





* run https://github.com/Automattic/woocommerce-payments/actions/workflows/e2e-pull-request.yml and confirm tests are passing on this branch

-------------------


**Post merge**
